src/query/interpreter.cpp

```diff
@@ -1861,6 +1870,36 @@ PreparedQuery PrepareIsolationLevelQuery(ParsedQuery parsed_query, const bool in
       RWType::NONE};
 }
 
+PreparedQuery PrepareAnalyticsModeQuery(ParsedQuery parsed_query, const bool in_explicit_transaction,
+                                        InterpreterContext *interpreter_context, Interpreter *interpreter) {
+  if (in_explicit_transaction) {
+    throw AnalyticsModeModificationInMulticommandTxException();
+  }
+
+  auto *analytics_mode_query = utils::Downcast<AnalyticsModeQuery>(parsed_query.query);
+  MG_ASSERT(analytics_mode_query);
+
+  const auto analytics_mode = ToStorageAnalyticsMode(analytics_mode_query->analytics_mode_);
+
+  auto callback = [analytics_mode_query, analytics_mode, interpreter_context, interpreter]() -> std::function<void()> {
+    switch (analytics_mode_query->analytics_mode_) {
+      case AnalyticsModeQuery::AnalyticsMode::ON:
+        return [interpreter_context, analytics_mode] { interpreter_context->db->SetAnalyticsMode(analytics_mode); };
+      case AnalyticsModeQuery::AnalyticsMode::OFF:
+        return [interpreter_context, analytics_mode] { interpreter_context->db->SetAnalyticsMode(analytics_mode); };
+    }
+  }();
```

Comment:
but these 2 switch statements have the same behaviour? can't you make a normal callback?